From da0881a336ffdd006432b20dc632ec212a07b1ac Mon Sep 17 00:00:00 2001
From: TheKit <nekit1000@gmail.com>
Date: Fri, 21 Apr 2023 14:35:24 +0500
Subject: [PATCH] (halium) never set FLAT_BINDER_FLAG_TXN_SECURITY_CTX flag

It breaks binder transactions when run on a kernel without
SELinux support.

Change-Id: I1521cc6bda7ad981f434e611a5503afeb2143d57
---
 libs/binder/Parcel.cpp       | 5 +++--
 libs/binder/ProcessState.cpp | 3 ++-
 2 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/libs/binder/Parcel.cpp b/libs/binder/Parcel.cpp
index 58b0b35..ea6bcf0 100644
--- a/libs/binder/Parcel.cpp
+++ b/libs/binder/Parcel.cpp
@@ -232,9 +232,10 @@ status_t Parcel::flattenBinder(const sp<IBinder>& binder) {
                 schedBits = schedPolicyMask(policy, priority);
             }
             obj.flags = FLAT_BINDER_FLAG_ACCEPTS_FDS;
-            if (local->isRequestingSid()) {
+            // Disabled for Halium
+            /*if (local->isRequestingSid()) {
                 obj.flags |= FLAT_BINDER_FLAG_TXN_SECURITY_CTX;
-            }
+            }*/
             if (local->isInheritRt()) {
                 obj.flags |= FLAT_BINDER_FLAG_INHERIT_RT;
             }
diff --git a/libs/binder/ProcessState.cpp b/libs/binder/ProcessState.cpp
index 4a01d81..18d5e68 100644
--- a/libs/binder/ProcessState.cpp
+++ b/libs/binder/ProcessState.cpp
@@ -193,7 +193,8 @@ bool ProcessState::becomeContextManager()
     AutoMutex _l(mLock);
 
     flat_binder_object obj {
-        .flags = FLAT_BINDER_FLAG_TXN_SECURITY_CTX,
+	// Disabled for Halium
+        /*.flags = FLAT_BINDER_FLAG_TXN_SECURITY_CTX,*/
     };
 
     int result = ioctl(mDriverFD, BINDER_SET_CONTEXT_MGR_EXT, &obj);
-- 
2.39.2

